app/models/case/base.rb

```diff
@@ -1030,7 +1030,11 @@ def set_managing_team
   end
 
   def set_number
-    self.number = next_number
+    self.number = if current_state == "rejected"
```

Comment:
The rejected state only affects Offender SARs, so it's probably best to only make this change in that class. You can override this method as it's a subclass.